### PR TITLE
Rationalize FIPS sources

### PR DESCRIPTION
**Pull request by t8m**
Avoid including QUIC related stuff in the FIPS sources. Also avoid including libssl headers in ssl3_cbc.c.

### CODE REVIEW

File: crypto/packet.c
Comment (levitte) [inline crypto/packet.c:14]: Wouldn't it be cleaner to have a separate packet_quic.c?

**Review by levitte — CHANGES_REQUESTED** [review]

File: crypto/packet.c
Comment (t8m) [reply crypto/packet.c:14]: I tried that but I've realized it is not possible given the put_quic_value() call in wpacket_intern_close().

**Comment by t8m** [comment]
Rebased to fix conflict.

**Review by paulidale — APPROVED** [review]
For 3.1 as well?

**Comment by t8m** [comment]
> For 3.1 as well?

This applies to master branch only. I'll review the fips sources on 3.1 after the batch backports are merged.

**Comment by paulidale** [comment]
I suspected that was the case.

**Comment by t8m** [comment]
Rebased to resolve trivial conflict in crypto/build.info. @paulidale still OK? Ping for second review.

**Comment by t8m** [comment]
Ping for the second review

**Review by mattcaswell — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
@mattcaswell @paulidale please reconfirm after rebase.

**Comment by paulidale** [comment]
Good here

**Comment by paulidale** [comment]
@mattcaswell ?

**Review by mattcaswell — APPROVED** [review]
Reconfirm

**Comment by t8m** [comment]
Merged to master branch. Thank you for the reviews.

**Pull request closed without merging**
